app/src/pages/inside/common/modals/editItemsModal/editItemsModal.jsx

```diff
@@ -0,0 +1,340 @@
+const ATTRIBUTE_DELETE = 'DELETE';
+
+const DESCRIPTION_LEAVE = 'LEAVE';
+const DESCRIPTION_UPDATE = 'UPDATE';
+const DESCRIPTION_CREATE = 'CREATE';
+
+const makeDescriptionOptions = (formatMessage) => [
+  {
+    value: DESCRIPTION_LEAVE,
+    label: formatMessage(messages.descriptionLeaveLabel),
+  },
+  {
+    value: DESCRIPTION_UPDATE,
+    label: formatMessage(messages.descriptionAddLabel),
+  },
+  {
+    value: DESCRIPTION_CREATE,
+    label: formatMessage(messages.descriptionReplaceLabel),
+  },
+];
+
+@withModal('editItemsModal')
+@injectIntl
+@reduxForm({
+  form: 'editItemsForm',
+})
+@formValues('descriptionAction', 'uniqueAttributes')
+@connect(
+  (state) => ({
+    currentProject: activeProjectSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+export class EditItemsModal extends Component {
+  static propTypes = {
+    data: PropTypes.shape({
+      items: PropTypes.array,
+      type: PropTypes.string,
+      fetchFunc: PropTypes.func,
+    }).isRequired,
+    initialize: PropTypes.func.isRequired,
+    currentProject: PropTypes.string.isRequired,
+    descriptionAction: PropTypes.string,
+    uniqueAttributes: PropTypes.array,
+    intl: intlShape.isRequired,
+    ...formPropTypes,
+  };
+
+  static defaultProps = {
+    descriptionAction: DESCRIPTION_LEAVE,
+    uniqueAttributes: [],
+  };
+
+  componentDidMount() {
+    const {
+      initialize,
+      data: { items },
+    } = this.props;
+    const attributes = this.getUniqueAndCommonAttributes();
+
+    initialize({
+      ids: items.map((item) => item.id),
+      commonAttributes: attributes[ATTRIBUTES_COMMON],
+      uniqueAttributes: attributes[ATTRIBUTES_UNIQUE],
+      attributes: [],
+      descriptionAction: DESCRIPTION_LEAVE,
+    });
+  }
+
+  onChangeCommonAttributes = (e, attributes, oldAttributes) => {
+    const { array } = this.props;
+    const saveHistory = (payload) => array.push('attributes', payload);
+
+    // Create attribute
+    if (attributes.length > oldAttributes.length) {
+      const createdAttribute = this.findAttribute(attributes, oldAttributes);
+
+      this.removeUniqueAttribute(createdAttribute);
+
+      saveHistory({
+        action: ATTRIBUTE_CREATE,
+        to: createdAttribute,
+      });
+
+      // Delete attribute
+    } else if (attributes.length < oldAttributes.length) {
+      const deletedAttribute = this.findAttribute(oldAttributes, attributes);
+
+      saveHistory({
+        action: ATTRIBUTE_DELETE,
+        from: deletedAttribute,
+      });
+
+      // Update attribute
+    } else {
+      const attributeBeforeUpdate = this.findAttribute(oldAttributes, attributes);
+      const updatedAttribute = this.findAttribute(attributes, oldAttributes);
+
+      this.removeUniqueAttribute(updatedAttribute);
+
+      saveHistory({
+        action: ATTRIBUTE_UPDATE,
+        from: attributeBeforeUpdate,
+        to: updatedAttribute,
+      });
+    }
+  };
+
+  getUniqueAndCommonAttributes = () => {
```

Comment:
I think it's better to move util functions to a separate file and add corresponding tests.

Reply:
Done.